nautobot/extras/models/models.py

```diff
@@ -517,6 +522,73 @@ def __str__(self):
     class Meta:
         ordering = ["name"]
 
+    def get_http_session(
+        self,
+        include_auth=True,
+        authorization_header="Authorization",
+        authorization_prefix="Bearer ",
+    ) -> requests.Session:
+        """Get a requests.Session object that has been initialized for this external integration.
+
+        The configuration will include SSL verification/CA file path options as well as optional
+        authentication parameters. If `include_auth` is `True` then the secrets group for this
+        external integration is obtained and either token or username/password authentication is
+        setup in the session headers. The session will prefer token authentication if the secrets
+        group has both token and username/password. If found, the token will be added to the
+        session headers using the `authorization_header` header key. The key value will be prefixed by the
+        `authorization_prefix` value.
+
+        If `include_auth` is `True` and no token is found, then username and password secrets
+        are retrieved and set for the session's basic auth values.
+
+        If `include_auth` is `True` and no valid secrets are found then a ValueError is raised.
+
+        Args:
+            include_auth (bool, optional): Whether or not to set basic auth or token auth
+                values in the session. Defaults to True.
+            authorization_header (str, optional): Used if `include_auth` is set. Sets the token header
+                key in the session. Defaults to "Authorization".
+            authorization_prefix (str, optional): Used if `include_auth` is set. Adds a prefix
+                to the token value in the HTTP request header. Defaults to "Bearer ".
+
+        Raises:
+            ValueError: If `include_auth` is set and no appropriate secrets are found.
+
+        Returns:
+            requests.Session: The session object with all the configured attributes.
+        """
+        session = requests.Session()
+        session.verify = self.ca_file_path or self.verify_ssl
+        if session.verify is False:
+            urllib3.disable_warnings(category=urllib3.exceptions.InsecureRequestWarning)
```

Comment:
Feels off to me that we disable this warning globally. Can it be disabled just in the context of the specific session isntead?

Reply:
I agree, this does feel a little off. I'll investigate if we can restrict it, the way I did it is based on what little documentation I could find. Perhaps we shouldn't do anything, if you've disable SSL verification you probably should get an obnoxious amount of warning messages.